src/test/java/edu/unc/lib/boxc/migration/cdm/services/RedirectMappingIndexServiceTest.java

```diff
@@ -0,0 +1,160 @@
+        testHelper.populateDescriptions("gilmer_mods1.xml");
+        testHelper.populateSourceFiles("276_182_E.tif", "276_183B_E.tif", "276_203_E.tif");
+
+        sipsService.generateSips(makeOptions());
+    }
+
+    @Test
+    public void indexingDoesNotHappenIfCollectionIsNotSubmitted() {
+        try {
+            indexService.indexMapping();
+        } catch (InvalidProjectStateException e) {
+            assertTrue("Unexpected message: " + e.getMessage(),
+                    e.getMessage().contains("Must submit the collection prior to indexing"));
+        }
+    }
+
+    @Test
+    public void redirectMappingTableIsPopulatedWithRightNumberOfRows() throws Exception {
+        addSipsSubmitted();
+        indexService.indexMapping();
+
+        Connection conn = indexService.openDbConnection();
+        try {
+            Statement stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery("select count(*) from redirect_mappings");
+            rs.next();
+            assertEquals("Incorrect number of rows in database", 3, rs.getInt(1));
+        } finally {
+            CdmIndexService.closeDbConnection(conn);
+        }
+    }
+
+    @Test
+    public void  redirectMappingIndexPopulatesColumnsCorrectly() throws Exception {
+        ArrayList<String> row1 = new ArrayList<>();
+        Path mappingPath = project.getRedirectMappingPath();
+        addSipsSubmitted();
+        indexService.indexMapping();
+
+        Reader reader = Files.newBufferedReader(mappingPath);
+        CSVParser originalParser = new CSVParser(reader, CSVFormat.DEFAULT
+                .withFirstRecordAsHeader()
+                .withHeader(RedirectMappingService.CSV_HEADERS)
+                .withTrim());
+
+        for (CSVRecord originalRecord : originalParser) {
+            row1.add(originalRecord.get(0)); // cdm_collection_id
+            row1.add(originalRecord.get(1)); // cdm_object_id
+            row1.add(originalRecord.get(2)); // boxc_work_id
+            row1.add(originalRecord.get(3)); // boxc_file_id
+            break; // just testing the first row
+        }
+
+        Connection conn = indexService.openDbConnection();
+        try {
+            Statement stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery("select cdm_collection_id, cdm_object_id, " +
+                    "boxc_work_id, boxc_file_id from redirect_mappings");
+            rs.next();
+            assertEquals("cdm_collection_id value isn't accurate", row1.get(0), rs.getString("cdm_collection_id"));
+            assertEquals("cdm_object_id value isn't accurate", row1.get(1), rs.getString("cdm_object_id"));
+            assertEquals("boxc_work_id value isn't accurate", row1.get(2), rs.getString("boxc_work_id"));
+            assertEquals("boxc_file_id value isn't accurate", row1.get(3), rs.getString("boxc_file_id"));
+        } finally {
+            CdmIndexService.closeDbConnection(conn);
+        }
+    }
+
```

Comment:
I would suggest having one more test to verify that compound objects work (cases where we have an entry without a File id) and that nulls are getting set, since if not that will mess up the queries on the redirect end. This can be setup with the following:
```
        testHelper.indexExportData(Paths.get("src/test/resources/keepsakes_fields.csv"), "export_compounds.xml");
        testHelper.getDescriptionsService().generateDocuments(true);
        testHelper.getDescriptionsService().expandDescriptions();
        testHelper.populateSourceFiles("nccg_ck_09.tif", "nccg_ck_1042-22_v1.tif",
                "nccg_ck_1042-22_v2.tif", "nccg_ck_549-4_v1.tif", "nccg_ck_549-4_v2.tif");
```
This is stolen from `StatusCommandIT`

Reply:
Okay, I've added a compound objects test that makes sure that the boxc_work_id is always populated. Let me know if I could have implemented that a better way!